http-clients/src/main/java/org/triplea/http/client/lobby/game/listing/GameListingClient.java

```diff
@@ -0,0 +1,57 @@
+package org.triplea.http.client.lobby.game.listing;
+
+import java.net.URI;
+import java.util.List;
+import lombok.AllArgsConstructor;
+import org.triplea.http.client.AuthenticationHeaders;
+import org.triplea.http.client.HttpClient;
+
+/**
+ * Http client for interacting with lobby game listing. Can be used to post, remove, boot, fetch and
+ * update games.
+ */
+@AllArgsConstructor
+public class GameListingClient {
+
+  public static final String BOOT_GAME_PATH = "/lobby/games/boot-game";
+  public static final String FETCH_GAMES_PATH = "/lobby/games/fetch-games";
+  public static final String KEEP_ALIVE_PATH = "/lobby/games/keep-alive";
+  public static final String POST_GAME_PATH = "/lobby/games/post-game";
+  public static final String UPDATE_GAME_PATH = "/lobby/games/update-game";
+  public static final String REMOVE_GAME_PATH = "/lobby/games/remove-game";
```

Comment:
Just a small side-note, if you were using HTTP methods besides `GET` and `POST` like `PUT`, `PATCH` and `DELETE`, this would allow to use the same uri string for multiple actions, which at least IMO makes it simpler to work with.
Example: `PUT /lobby/game`, `PATCH /lobby/game` and `DELETE /lobby/game` vs. `POST /lobby/games/post-game`, `POST /lobby/games/update-game` and `POST /lobby/games/remove-game`.

Of course those methods have different implications like the caching mechanisms that come with `GET` and make promises to have or not have side-effects, or to be indempotent, so that would need checking, but it might make it easier to make assumptions about the API in the future.
Obviously the overall benefit in this case is rather small, and REST really only comes to shine on larger use-cases, but I still think it would be nice to use this here.

Reply:
PUT usually implies a file upload, right?

REST AFAIK also implies self-documentation. Similar to the github api for example, a request to the root URL tells you which other URLs are available. I think what we'd done here is simply JSON over HTTP. Using status codes well I think is a win, proper use the HTTP verbs, I'd agree the benefit is for us kinda marginal.

On the other hand, I do want to see an example of HTTP PUT with file upload. There is a lot of utility in that. We could replace the forum poster with a feign client and open up a lot of options like 'cloud save-games' and upload of map files and images.

Reply:
AFAIK `PUT` only implies the creation of a resource. What defines such a resource and from what kind of data it is created isn't defined further.
So creating a game entry could be seen as creating a resource.

Reply:
We could do something like:

```
POST /games
PUT /games/<GAME_ID>
GET /games/<GAME_ID>
PATCH /games/<GAME_ID>
```

The difference between PUT and PATCH seems ambiguous. Passing the 'game_id' through parameter or URL string is not very much different. I'm not sure there is that much value for having more faithful adherence to REST. We're not necessarily doing REST here.